win32ss/user/ntuser/timer.c

```diff
@@ -185,28 +185,18 @@ IntSetTimer( PWND Window,
   LARGE_INTEGER DueTime;
   DueTime.QuadPart = (LONGLONG)(-97656); // 1024hz .9765625 ms set to 10.0 ms
 
-#if 0
-  /* Windows NT/2k/XP behaviour */
-  if (Elapse > USER_TIMER_MAXIMUM)
-  {
-     TRACE("Adjusting uElapse\n");
-     Elapse = 1;
-  }
-#else
-  /* Windows XP SP2 and Windows Server 2003 behaviour */
-  if (Elapse > USER_TIMER_MAXIMUM)
-  {
-     TRACE("Adjusting uElapse\n");
-     Elapse = USER_TIMER_MAXIMUM;
-  }
-#endif
-
   /* Windows 2k/XP and Windows Server 2003 SP1 behaviour */
   if (Elapse < USER_TIMER_MINIMUM)
   {
-     TRACE("Adjusting uElapse\n");
+     TRACE("Adjusting uElapse to USER_TIMER_MINIMUM\n");
      Elapse = USER_TIMER_MINIMUM; // 1024hz .9765625 ms, set to 10.0 ms (+/-)1 ms
   }
+  else if (Elapse > USER_TIMER_MAXIMUM)
+  {
+     /* Windows XP SP2 and Windows Server 2003 new behaviour */
+     TRACE("Adjusting uElapse to USER_TIMER_MAXIMUM\n");
+     Elapse = USER_TIMER_MAXIMUM;
+  }
```

Comment:
Not sure we really care about these traces, in which case doing a
```Elapse = min(max(Elapse, USER_TIMER_MINIMUM), USER_TIMER_MAXIMUM);```
would be more "straightforward".

Reply:
TRACE: I don't know, so I am not removing them (yet).
min/max: See reply to other comment.